src/Config.php

```diff
@@ -168,47 +166,45 @@ public function getTypePreference(string $type, ?string $context = null) : ?stri
         }
 
         $preference = $this->preferences[$type];
-        return ($preference != '') ? (string)$preference : null;
+        return $preference !== '' ? (string) $preference : null;
     }
 
     /**
      * {@inheritDoc}
+     *
      * @see Laminas\Di\ConfigInterface::getTypePreferencesForClass()
      */
-    private function getTypePreferenceForClass(string $type, ?string $context) : ?string
+    private function getTypePreferenceForClass(string $type, ?string $context): ?string
     {
         if (! isset($this->types[$context]['preferences'][$type])) {
             return null;
         }
 
         $preference = $this->types[$context]['preferences'][$type];
-        return ($preference != '') ? (string)$preference : null;
+        return $preference !== '' ? (string) $preference : null;
```

Comment:
This introduces a BC break: When some Stringable (i.e. a class with a `__toString()` method) is placed in the config it will now change behavior. Prior this change it would be accepted and stringified. After this it becomes `null`.

Reply:
This one was also made by php-cs. check if $preference is a class and has __toString and call __toString by ourselves?

Reply:
@boesing and @tux-rampage this problem still has to be fixed!
What do you guys think how to solve it?

Reply:
@gennadigennadigennadi You can run `vendor/bin/phpcs -s` and check which rule is trying to change it. When you know, you can add `// phpcs:ignore some.phpcs.rule` in front of it so it won't get changed.